src/aspire/source/xform.py

```diff
@@ -126,7 +133,25 @@ def __init__(self, factor):
         self.multipliers = factor
 
     def _forward(self, im, indices):
-        return im * self.multipliers[indices]
+        if self.multipliers.size == 1:
```

Comment:
If  `self.multipliers` was the correct type (see `__init__` comment above), this would not be needed. The takeaway is that do whatever is needed ahead of time so you can take advantage of numpy broadcasting and not have if/else conditions like these.

Reply:
I am still working on removing the checking of dimension. `self.multipliers` is Numpy array now but it can be size of 1 and size of N. The real operation size should be `len(indices)`. Not sure how to fit it without explicit converting.